tasks/workflow/step.go

```diff
@@ -261,19 +261,33 @@ func (s *step) runActivity(wfCtx context.Context, cfg config.Configuration, depl
 			// Need to publish INITIAL status before RUNNING one with operationName set
 			s.publishInstanceRelatedEvents(wfCtx, deploymentID, instanceName, eventInfo, tasks.TaskStepStatusINITIAL, tasks.TaskStepStatusRUNNING)
 		}
+
+		executorDelegateLabelDeployment := metrics.Label{
+			Name:  "Deployment",
+			Value: deploymentID,
+		}
+		executorDelegateLabelName := metrics.Label{
+			Name:  "Name",
+			Value: delegateOp,
+		}
+		executorDelegateLabelNode := metrics.Label{
+			Name:  "Node",
+			Value: nodeType,
+		}
+		executorDelegateLabels := []metrics.Label{executorDelegateLabelDeployment, executorDelegateLabelName, executorDelegateLabelNode}
 		err = func() error {
-			defer metrics.MeasureSince(metricsutil.CleanupMetricKey([]string{"executor", "delegate", deploymentID, nodeType, delegateOp}), time.Now())
+			defer metrics.MeasureSinceWithLabels(metricsutil.CleanupMetricKey([]string{"executor", "delegate"}), time.Now(), executorDelegateLabels)
```

Comment:
The metric is called `executor.delegate` here, shouldn't it be called something like `executor.delegate.duration` to describe it provides the duration of an execution ?